test/multisigwallet/submit_requirement_change.js

```diff
@@ -0,0 +1,365 @@
+
+            assert.isNotOk(
+                await multisig.confirmations(transactionID, registeredWallet1),
+                'Non submitter should not confirm.',
+            );
+
+            assert.isOk(
+                (await multisig.transactionCount.call()).eqn(1),
+                'Transaction count should be increased by one.',
+            );
+
+            const transaction = await multisig.transactions.call(transactionID);
+
+            assert.strictEqual(
+                transaction.destination,
+                multisig.address,
+            );
+
+            assert.strictEqual(
+                transaction.data,
+                generateRequirementChangeData(newRequired),
+            );
+
+            assert.strictEqual(
+                transaction.executed,
+                true,
+            );
+        });
+
+        it('Submit => Confirm', async () => {
+            const required = 2;
+
+            const registeredWallet0 = accounts[0];
+            const registeredWallet1 = accounts[1];
+            const registeredWallet2 = accounts[2];
+
+            const wallets = [
+                registeredWallet0, registeredWallet1, registeredWallet2,
+            ];
+
+            const multisig = await MultiSigWallet.new(wallets, required);
+
+            const newRequired = 3;
+
+            await multisig.submitRequirementChange(
+                newRequired,
+                {
+                    from: registeredWallet0,
+                },
+            );
+
+            const transactionID = 0;
+
+            assert.isOk(
+                (await multisig.required.call()).eqn(required),
+                'As required is equal to 2, the transaction would not be '
+                + 'executed in the same call.',
+            );
+
+            assert.isOk(
+                await multisig.confirmations(transactionID, registeredWallet0),
+                'Submitter should also confirm.',
+            );
+
+            assert.isNotOk(
+                await multisig.confirmations(transactionID, registeredWallet1),
+                'Non submitter should not confirm.',
+            );
+
+            assert.isNotOk(
+                await multisig.confirmations(transactionID, registeredWallet2),
+                'Non submitter should not confirm.',
+            );
+
+            assert.isOk(
+                (await multisig.transactionCount.call()).eqn(1),
+                'Transaction count should be increased by one.',
+            );
+
+            const transaction = await multisig.transactions.call(transactionID);
+
+            assert.strictEqual(
+                transaction.destination,
+                multisig.address,
+            );
+
+            assert.strictEqual(
+                transaction.data,
+                generateRequirementChangeData(newRequired),
+            );
+
+            assert.strictEqual(
+                transaction.executed,
+                false,
+            );
+        });
+    });
+});
```

Comment:
Following points to be considered for all the files :-
1. Remove commas where it is not required.I have commented to remove it on some places.But,it needs to be changed all the places.
2. There are test cases under "Storage".Provide more descriptive message.
3. While confirming and revoking transaction,instead of hardcoding the transactionid,get the transaction id using "call".
4. Declare common variables used in the test cases at top of the file after requiring files.Use let keyword instead of const to declare variables for the test cases.

  
Test cases to be added :-
1. When removing wallet,if the wallet count reaches below requirement.(I believe we should fire  requirementChanged event).
2. Transaction to remove non-existent wallet.
3. For replacing wallet,make old wallet as null.Also,submit transaction where old and new wallet is null.

Let me know in case of any query.

Reply:
1. Remove commas where it is not required.I have commented to remove it on some places.vBut,it needs to be changed all the places.
    -- The ending comma is according to Airbnb coding style (we do not follow it strictly, however, we should set up eslint and use `eslint --fix` for formatting. This would keep code cleaner and will make to use a newer constructs of javascript).

2. There are test cases under "Storage". Provide more descriptive message.
    -- Mostly they are testing setting/updating storage. Could you please let me know which ones need improvements. Thx.

3. While confirming and revoking transaction, instead of hardcoding the transactionID, get the transaction id using "call".
    -- Done (including other files).

4. Declare common variables used in the test cases at top of the file after requiring files. Use let keyword instead of const to declare variables for the test cases.
    -- I do not agree that usage of test global variables make test readable and maintainable. Currently the context of each test case (`it`) is visible within it's scope and is not affected by other `it` sections.

Test cases to be added :-
1. When removing wallet, if the wallet count reaches below requirement. (I believe we should fire  requirementChanged event).
   - Done. Regarding requirementChanged is not going to be emitted. TransactionExecucution* events are going to be fired.
2. Transaction to remove non-existent wallet.
    -- Done.
3. For replacing wallet, make old wallet as null. Also, submit transaction where old and new wallet is null.
    -- We do not need to check old wallet being null case as the check that old wallet is registered is enough (there is no way to register null wallet).
    -- There is a check that test expects to fail if new wallet is null. Old wallet to be null in this case is the same as in above point.

Reply:
1. It's checking the storage changes/updates that happens once the tested function calls happen. Function might have different changes based on previous state. The `it` sections within `contract` sections are defining those scenarios. In each `it` section within `contract` "Storage" we check all possible storage values that function might change. So it's not about one storage variable change.
2. Because it's highly valued by JS community (another one is Google). They have an extended, well written and maintained documentation. There are many tools one could use for linting and formatting against Airbnb coding style. It's nice to have one in our codebase :).